### PR TITLE
feat(building-comparison): add Microsoft Building Footprints dataset

### LINKED ISSUE #761: building-comparison: add coverage for each dataset to a FeatureCollection
So we can show the coverage of the different datasets in the dashboard

### PR DESCRIPTION
**Pull request by Gigaszi**
Please add the **type of change** as label. If your PR is not ready for review and merge, please add `🚧 ` to the PR title.

### Description
Please add a clear and concise description of what your PR solves.

### Corresponding issue
Closes #761 

### New or changed dependencies
-

### Checklist
- [ ] I have updated my branch to `main` (e.g. through `git rebase main`)
- [ ] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-api/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-api/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [ ] I have commented my code
- [ ] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-api/blob/main/docs/development_setup.md#tests)
- [ ] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-api/blob/main/CHANGELOG.md)
- [ ] Add attribution for microsoft buildings

Please check all finished tasks. If some tasks do not apply to your PR, please cross their text out (by using `~...~`) and remove their checkboxes.

### CODE REVIEW

File: ohsome_quality_api/geodatabase/select_check_reference_coverage.sql [hunk 1]
```diff
@@ -7,6 +7,6 @@ SELECT
     ST_Area (ST_Intersection (bpoly.geom, coverage.geom)) / ST_Area (bpoly.geom) as area_ratio
 FROM
     bpoly,
-    eubucco_v0_1_coverage_simple coverage
+    {coverage_name} coverage
```
Comment (matthiasschaub) [inline ohsome_quality_api/geodatabase/select_check_reference_coverage.sql:10]:
```suggestion
    {table_name} coverage
```

File: ohsome_quality_api/geodatabase/get_coverage_intersection.sql [hunk 2]
```diff
@@ -6,6 +6,6 @@ SELECT
     ST_AsGeoJSON (ST_Intersection (bpoly.geom, coverage.geom)) AS geom
 FROM
     bpoly,
-    eubucco_v0_1_coverage_simple coverage
+    {coverage_name} coverage
```
Comment (matthiasschaub) [inline ohsome_quality_api/geodatabase/get_coverage_intersection.sql:9]:
```suggestion
    {table_name} coverage
```

File: ohsome_quality_api/indicators/base.py [hunk 3]
```diff
@@ -102,22 +102,26 @@ def attribution(cls) -> str:
         return get_attribution(["OSM"])
 
     @classmethod
-    async def coverage(cls, inverse=False) -> Polygon | MultiPolygon:
+    async def coverage(cls, inverse=False) -> list[Feature]:
         """Return coverage geometry. Default is global coverage."""
```
Comment (matthiasschaub) [inline ohsome_quality_api/indicators/base.py:106]:
```suggestion
        """Return coverage geometries. Default is global coverage."""
```

File: ohsome_quality_api/indicators/base.py [hunk 4]
```diff
@@ -102,22 +102,26 @@ def attribution(cls) -> str:
...
         if inverse is False:
```
Comment (matthiasschaub) [inline ohsome_quality_api/indicators/base.py:107]:
```suggestion
        if inverse:
            return [Feature(Polygon(coordinates=[]))]
```

File: tests/integrationtests/api/test_metadata_indicators.py [hunk 5]
```diff
@@ -86,7 +86,7 @@ def test_coverage_default(client):
     }
 
 
-@pytest.mark.skip(reason="Depends on database")
+# @pytest.mark.skip(reason="Depends on database")
```
Comment (matthiasschaub) [inline tests/integrationtests/api/test_metadata_indicators.py:89]:
```suggestion
@pytest.mark.skip(reason="Depends on database")
```

File: tests/unittests/test_indicators_definitions.py [hunk 6]
```diff
@@ -9,14 +9,14 @@
...
+def mock_select_coverage(class_mocker, feature_germany_berlin):
     async def side_effect(*args, **kwargs):
-        inverse = args[0]
+        inverse = args[1]
 
         if inverse:
             return [
-                {
-                    "geom": geojson.dumps(
+                str(
```
Comment (matthiasschaub) [inline tests/unittests/test_indicators_definitions.py:18]: Should not be necessary

File: tests/unittests/test_indicators_definitions.py [hunk 7]
```diff
@@ -29,14 +29,14 @@ async def side_effect(*args, **kwargs):
                             ]
                         )
                     )
-                }
+                )
             ]
         else:
-            return [{"geom": geojson.dumps(Polygon(coordinates=[]))}]
+            return [str(geojson.dumps(Polygon(coordinates=[])))]
```
Comment (matthiasschaub) [inline tests/unittests/test_indicators_definitions.py:35]: Same as above

File: ohsome_quality_api/geodatabase/client.py [hunk 8]
```diff
@@ -69,50 +69,74 @@ async def get_shdi(bpoly: Feature | FeatureCollection) -> list[Record]:
         return await conn.fetch(query, geom)
 
 
-async def get_building_area(bpoly: Feature) -> list[Record]:
+async def get_building_area(bpoly: Feature, table_names: list[str]) -> list[Record]:
```
Comment (matthiasschaub) [inline ohsome_quality_api/geodatabase/client.py:72]:
```suggestion
async def get_building_area(bpoly: Feature, table_names: Literal["EUBUCCO", "Microsoft-Buildings"]) -> list[Record]:
```

Comment (matthiasschaub) [reply ohsome_quality_api/geodatabase/client.py:72]: check strings

File: ohsome_quality_api/geodatabase/client.py [hunk 9]
```diff
@@ -69,50 +69,74 @@ async def get_shdi(bpoly: Feature | FeatureCollection) -> list[Record]:
...
     """Get area of building footprints for a bounding polygon."""
+    area_list = []
```
Comment (matthiasschaub) [inline ohsome_quality_api/geodatabase/client.py:74]: dict instead of list? then its easier to differentiate.

Comment (matthiasschaub) [reply ohsome_quality_api/geodatabase/client.py:74]: remember to change return type

File: ohsome_quality_api/geodatabase/client.py [hunk 10]
```diff
@@ -69,50 +69,74 @@ async def get_shdi(bpoly: Feature | FeatureCollection) -> list[Record]:
...
         query = file.read()
-    if inverse:
-        table_name = "eubucco_v0_1_coverage_inversed"
-    else:
-        table_name = "eubucco_v0_1_coverage_simple"
-    query = query.format(table_name=table_name)
+    geom = str(bpoly.geometry)
     async with get_connection() as conn:
-        return await conn.fetch(query)
+        return await conn.fetch(query, geom)
```
Comment (matthiasschaub) [inline ohsome_quality_api/geodatabase/client.py:95]: relict?

File: ohsome_quality_api/geodatabase/client.py [hunk 11]
```diff
@@ -69,50 +69,74 @@ async def get_shdi(bpoly: Feature | FeatureCollection) -> list[Record]:
...
 
 
-async def get_eubucco_coverage_intersection_area(bpoly: Feature) -> list[Record]:
+async def get_reference_coverage(table_names: list, inverse: bool) -> list[str]:
```
Comment (matthiasschaub) [inline ohsome_quality_api/geodatabase/client.py:98]: use literal

File: ohsome_quality_api/geodatabase/client.py [hunk 12]
```diff
@@ -69,50 +69,74 @@ async def get_shdi(bpoly: Feature | FeatureCollection) -> list[Record]:
...
+    """Get reference coverage for a bounding polygon."""
+    coverage_list = []
```
Comment (matthiasschaub) [inline ohsome_quality_api/geodatabase/client.py:100]: dict instead of list?

File: ohsome_quality_api/geodatabase/client.py (hunk 9 above)
Comment (hn437) [reply ohsome_quality_api/geodatabase/client.py:77]: Is this function used anywhere else than in the test 'test_get_building_area' from 'test_geodatabase'? Should the test rather use function 'get_reference_building_area' from indicator or should the indicator use the function from the client, but both doesn't make sense?

**Pull request closed without merging**